production/tanka/grafana-agent/v1/lib/deployment.libsonnet

```diff
@@ -26,25 +31,61 @@ local agent = import '../internal/agent.libsonnet';
 
     local has_prometheus_config = std.objectHasAll(self, '_prometheus_config'),
     local has_prometheus_instances = std.objectHasAll(self, '_prometheus_instances'),
+    local has_tempo_config = std.objectHasAll(self, '_tempo_config'),
+    local has_sampling_strategies = std.objectHasAll(self, '_tempo_sampling_strategies'),
 
     config:: {
       server: {
         log_level: 'info',
         http_listen_port: 8080,
       },
-
-      prometheus:
-        if !has_prometheus_config then {}
-        else this._prometheus_config {
-          configs:
-            if has_prometheus_instances
-            then this._prometheus_instances
-            else [],
+    } + (
+      if has_prometheus_config 
+      then {
+        prometheus:
+          this._prometheus_config {
+            configs:
+              if has_prometheus_instances
+              then this._prometheus_instances
+              else [],
+          },
+      }
+      else {}
+    ) + (
+      if has_tempo_config then {
+        tempo: {
+          configs: [this._tempo_config {
+            name: 'default',
+          }],
         },
-    },
+      }
+      else {}
+    ),
 
     agent:
       agent.newAgent(name, namespace, self._images.agent, self.config, use_daemonset=false) +
-      agent.withConfigHash(self._config_hash),
+      agent.withConfigHash(self._config_hash) + (
+        if has_tempo_config
+        then 
+          {
+            container+:: container.withEnvMixin([
+              k.core.v1.envVar.fromFieldPath('HOSTNAME', 'spec.nodeName'),
+            ]),
```

Comment:
Hmm, I kind of think we should have this regardless if tempo configs are defined. It's necessary for so much stuff (logs, integrations, etc)

Reply:
Good call, moved it out of the `if` clause. Note that this adds it to `agent-bare.yaml` as well (which doesn't really matter)